### PR TITLE
Enable 64-bit build with VC2015

### PR DESCRIPTION
**Pull request by atilaneves**
There is currently no top-level win64 makefile nor does dmd currently build with VS2015. This patch addresses both these issues.

### CODE REVIEW

File: src/ddmd/backend/gloop.c [hunk 1]
```diff
@@ -1034,7 +1034,7 @@ STATIC void markinvar(elem *n,vec_t rd)
         case OPcolon:
         case OPcolon2:
                 tmp = vec_clone(rd);
-                switch (el_returns(n->E1) * 2 | el_returns(n->E2))
+                switch (int(el_returns(n->E1) * 2) | int(el_returns(n->E2)))
```
Comment (John-Colvin) [inline src/ddmd/backend/gloop.c]: only need second cast

File: src/ddmd/backend/gother.c [hunk 2]
```diff
@@ -219,7 +219,7 @@ STATIC void conpropwalk(elem *n,vec_t IN)
         if (op == OPcolon || op == OPcolon2)
         {
                 L = vec_clone(IN);
-                switch (el_returns(n->E1) * 2 | el_returns(n->E2))
+                switch (int(el_returns(n->E1) * 2) | int(el_returns(n->E2)))
```
Comment (John-Colvin) [inline src/ddmd/backend/gother.c]: only need second cast

**Comment by WalterBright** [comment]
I thought this was already building with VS? @rainers ?

File: src/ddmd/backend/gflow.c [hunk 3]
```diff
@@ -323,7 +323,7 @@ STATIC void accumrd(vec_t GEN,vec_t KILL,elem *n)
                 rdelem(&Gl,&Kl,n->E1);
                 rdelem(&Gr,&Kr,n->E2);
 
-                switch (el_returns(n->E1) * 2 | el_returns(n->E2))
+                switch (el_returns(n->E1) * 2 | int(el_returns(n->E2)))
```
Comment (WalterBright) [inline src/ddmd/backend/gflow.c]: This looks strange - why is the cast needed?

**Comment by rainers** [comment]
> I thought this was already building with VS? @rainers ?

It does, but building through the makefile enables "warnings as errors".

File: src/ddmd/backend/gflow.c (hunk 3 above)
Comment (rainers) [reply src/ddmd/backend/gflow.c]: el_returns returns a boolean which causes:
warning C4805: '|' : unsafe mix of type 'int' and type 'bool' in operation

File: src/vcbuild/msvc-dmc.d [hunk 4]
```diff
@@ -19,6 +21,15 @@ int main(string[] args)
     newArgs ~= `/Ivcbuild`;
     newArgs ~= `/Iddmd\root`;
     newArgs ~= `/FIwarnings.h`;
+
+    enum vs2015 = "Microsoft Visual Studio 14.0";
+    if (environment.get("VCINSTALLDIR", "").canFind(vs2015) || environment.get("MSVC_CC", "").canFind(vs2015))
```
Comment (rainers) [inline src/vcbuild/msvc-dmc.d]: The check can fail if the installation path is not the default. Checking the version directly using VisualStudioVersion (also set via vcvarsall.bat) might be less brittle.

Comment (atilaneves) [reply src/vcbuild/msvc-dmc.d]: Just pushed changes to do that. Thanks for the comment, I didn't know about `%VisualStudioVersion%`.

File: src/vcbuild/msvc-dmc.d [hunk 5]
```diff
@@ -19,6 +21,14 @@ int main(string[] args)
     newArgs ~= `/Ivcbuild`;
     newArgs ~= `/Iddmd\root`;
     newArgs ~= `/FIwarnings.h`;
+
+    if (environment.get("VisualStudioVersion", "").to!double >= 14.0)
```
Comment (rainers) [inline src/vcbuild/msvc-dmc.d]: I'm not sure digger really uses the vcvarsall batches, so defaulting to version 10.0 might be better for backward compatibility. AFAIK to!double will cause an exception on empty input.

Comment (atilaneves) [reply src/vcbuild/msvc-dmc.d]: The autotester must, I was relying on that to catch any problems. I'll change as per your comment, it's much better that way anyway.

Comment (rainers) [reply src/vcbuild/msvc-dmc.d]: Thanks.

The autotester doesn't build dmd with VS, but with dmc. So no errors here will be caught.

File: src/ddmd/backend/gflow.c (hunk 3 above)
Comment (WalterBright) [reply src/ddmd/backend/gflow.c]: ok

**Pull request merged**